pkg/util/signal/signal_posix.go

```diff
@@ -26,22 +31,47 @@ import (
 	"go.uber.org/zap"
 )
 
-// SetupSignalHandler setup signal handler for TiDB Server
-func SetupSignalHandler(shutdownFunc func()) {
-	usrDefSignalChan := make(chan os.Signal, 1)
+// it's the same function used by "/debug/pprof/goroutine"
+// see https://github.com/golang/go/blob/e515ef8bc271f632bb2ebb94e8e700ab67274268/src/runtime/pprof/pprof.go#L750-L769
+func getGoroutineStacks() string {
+	// We don't know how big the buffer needs to be to collect
+	// all the goroutines. Start with 1 MB and try a few times, doubling each time.
+	// Give up and use a truncated trace if 64 MB is not enough.
+	buf := make([]byte, 1<<20)
+	for i := 0; ; i++ {
```

Comment:
seems `i` is not used?

Reply:
copied from golang source code, will keep it as long as ci pass